### PR TITLE
Changes the log to log every epoch

### PR DESCRIPTION
**Pull request by nikk-nikaznan**
The `.log` function cannot force the api to log for every epoch even when set `on_epoch=True` and `on_step=False`.
[Seems like unresolved issue](https://github.com/Lightning-AI/pytorch-lightning/issues/3228).

Instead we do `.log_metric` and set `step=self.current_epoch`. The x-axis will say step but it is in epoch.

Few other small changes:

- log model_filename in the mlflow
- options for fast debugging:

1. `fast_dev_run` only run for one batch and one epoch.
2. `limit_train_batches` run every epochs with limited percentage of training data.

### CODE REVIEW

File: crabs/detection_tracking/train_model.py [hunk 1]
```diff
@@ -36,6 +36,7 @@ class DectectorTrain:
     """
 
     def __init__(self, args):
+        self.args = args
```
Comment (sfmig) [inline crabs/detection_tracking/train_model.py:39]: fantastic additions! 🚀 

Just about the syntax:
I was wondering if we could have something more descriptive than `args`. Maybe:
```
 def __init__(self, args):
        [...]
	self.debug_fast_dev_run = args.fast_dev_run
	self.debug_limit_train_batches = args.limit_train_batches
	self.load_config_yaml()
```

Pros: it makes it more explicit what we are passing (hence more readable), it would be consistent with the rest of the `args` we have, and we avoid repetition (as it is now, we have `self.args.accelerator` and `self.accelerator` right?). Cons: less flexible when we add new args. But I think pros outweight cons.

File: crabs/detection_tracking/train_model.py [hunk 2]
```diff
@@ -119,14 +120,17 @@ def train_model(self):
             max_epochs=self.config["num_epochs"],
             accelerator=self.accelerator,
             logger=mlf_logger,
+            fast_dev_run=self.args.fast_dev_run,
+            limit_train_batches=self.args.limit_train_batches,
```
Comment (sfmig) [inline crabs/detection_tracking/train_model.py:124]:
```suggestion
            fast_dev_run=self.debug_fast_dev_run,
            limit_train_batches=self.debug_limit_train_batches,
```

Comment (sfmig) [reply crabs/detection_tracking/train_model.py:124]: Question: do we want these parameters in the config instead? I personally like them as CLI arguments but I'm not sure we have a clear distinction for what goes in config and what doesn't... 

The big advantage of putting it in the config is that all the config is dumped into an MLflow log. Could we also dump the CLI arguments to an MLflow log? I think this would be very useful. We can discuss this offline or open an issue.

File: crabs/detection_tracking/train_model.py [hunk 3]
```diff
@@ -119,14 +120,17 @@ def train_model(self):
...
         )
 
         # Run training
         trainer.fit(lightning_model, data_module)
 
         # Save model if required
         if self.config["save"]:
-            save_model(lightning_model)
+            model_filename = save_model(lightning_model)
+            mlf_logger.log_hyperparams({"model_filename": model_filename})
```
Comment (sfmig) [inline crabs/detection_tracking/train_model.py:133]: fantastic saving the model name ✨

File: crabs/detection_tracking/train_model.py [hunk 4]
```diff
@@ -191,6 +195,20 @@ def train_parse_args(args):
         default=42,
         help="seed for dataset splits",
     )
+    parser.add_argument(
+        "--fast_dev_run",
+        action="store_true",
+        help="option to run only one batch and one epoch for debugging",
```
Comment (sfmig) [inline crabs/detection_tracking/train_model.py:201]:
```suggestion
        help="Debugging option to run training for one batch and one epoch",
```

File: crabs/detection_tracking/train_model.py [hunk 5]
```diff
@@ -191,6 +195,20 @@ def train_parse_args(args):
...
+    )
+    parser.add_argument(
+        "--limit_train_batches",
+        type=float,
+        default=1.0,
+        help=(
+            "option to run smaller training number per epoch for debugging."
```
Comment (sfmig) [inline crabs/detection_tracking/train_model.py:208]:
```suggestion
            "Debugging option to run training on a fraction of the training set."
```

**Review by sfmig — APPROVED** [review]
thanks, looking neat! 🚀 some suggestions and one question

File: crabs/detection_tracking/train_model.py (hunk 2 above)
Comment (nikk-nikaznan) [reply crabs/detection_tracking/train_model.py:124]: I think technically, things like hyper parameters in config, things we need to change quickly and many times in cli. Like debugging so we don't keep changing the config files for example from True to False too much. I don't know really, I think it's preference as well.

Comment (nikk-nikaznan) [reply crabs/detection_tracking/train_model.py:124]: I think we keep as it is for now, we can always move things around especially if we see thing growing too long
